src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -670,7 +653,61 @@ private static FunctionCompletedMessage CreateCompletedMessage(FunctionStartedMe
                 ParameterLogBlob = startedMessage.ParameterLogBlob
             };
         }
-                
+
+        // Called very early when function is started; before arguments are bound. 
+        private Task NotifyPreBindAsync(FunctionInstanceLogEntry fastItem)
+        {
+            fastItem.LiveTimer = Stopwatch.StartNew();
+
+            if (_functionEventCollector == null)
+            {
+                return Task.CompletedTask;
+            }            
+
+            // Log pre-bind event. 
+            return _functionEventCollector.AddAsync(fastItem);
```

Comment:
It feels like this `_functionEventCollector` really isn't much of an `AsyncCollector` anymore. Now, if you want to implement this, you decipher the current state of the `FunctionInstanceLogEntry` to determine where in the invocation life cycle you are. `AddAsync` and `FlushAsync` don't really seem to be enough to describe what's happening.

Did you think about making these `Notify` methods virtual so that we could override them in Script to know for sure what event was being fired?